admin/listPurchases/listView.html

```diff
@@ -100,51 +100,67 @@
     });
 });
 
-const amountOfShownSubpages = 5;
 const entriesPerSubpage = 20;
+const amountOfShownSubpages = 5;
 let amountOfSubpages;
 
 function generateSubPageNavigation(){
-    if(orderSearchBar.value){
-        amountOfSubpages = Math.ceil(amountOfEntries/ entriesPerSubpage);
-    }else{
-        amountOfSubpages = Math.ceil(amountOfEntries / entriesPerSubpage);
-    }
+    amountOfSubpages = Math.ceil(amountOfEntries/ entriesPerSubpage);
 
-    if(currentPageIndex >= amountOfSubpages){
-        currentPageIndex = amountOfSubpages - 1;
-    } 
+    const extraNavNeeded = amountOfSubpages > amountOfShownSubpages;
 
     let startIndexToRender = 0;
     let subpageNavigation = [];
 
-    const extraNavNeeded = amountOfSubpages > amountOfShownSubpages;
+    if(extraNavNeeded){
+        startIndexToRender = ShiftedStartIndexToRender();
+        addBackwardsExtraNav(subpageNavigation);
```

Comment:
When you pass the array you are building *into* functions it obscures what is happening: *anything* might happen in there!

For clarity, I would prefer something like this, building the array out of the relevant parts in one go:

```js
const subpageNavigation = [
  ...(extraNavNeeded ? makeBackwardsExtraNav() : []),
  ...makeDefaultNav(startIndexToRender),
  ...(extraNavNeeded ? makeForwardsExtraNav() : []),
];
```

This shows that subpageNavigation is an array consisting of sequential subarrays.